addons/mail/static/src/messaging/entity/thread_viewer/thread_viewer.js

```diff
@@ -191,14 +218,34 @@ function ThreadViewerFactory({ Entity }) {
             });
 
             if (thread && this.thread !== thread) {
+                this._stopLoaderTimeout();
                 this.link({ thread });
                 if (!this.threadCache.isLoaded && !this.threadCache.isLoading) {
                     this.threadCache.loadMessages();
                 }
             }
             if (this.threadCache !== prevThreadCache) {
+                this._stopLoaderTimeout();
                 this._writeComponentHint('change-of-thread-cache');
             }
+
+            if (!this.thread) {
+                this._stopLoaderTimeout();
+                this.isShowingLoading = false;
+            } else {
+                const threadCache = this.threadCache;
+                if (threadCache.isLoading) {
+                    if (!this._loaderTimeout) {
+                        this._loaderTimeout = setTimeout(() => {
+                            this._stopLoaderTimeout();
+                            this.isShowingLoading = threadCache.isLoading;
+                        }, 400);
```

Comment:
`this._loaderTimeout = setTimeout(() => this.isShowingLoading = true, 400);`

Reply:
Mmh what if the timeout is executed when the thread is loaded ? That'll show the loading or it should not, no ?

Reply:
Showing the loading happens when viewer detects cache is loading and it is still loading after the timeout. Changes of thread or thread cache or when cache becomes loaded, any timeout is aborted and `isShowingLoading` is unset.

Reply:
There is no way the timeout is cancelled after 399s, and the loader is hidden but timeout function is still executed showing the loading ? If there is no way such thing could happen, then I agree with you else I think it could be a problem as the loading will be shown "forever" (would need a _update or a thread loaded event to trigger the removal of the loading)

Reply:
Everything is single-threaded and a `clearTimeout()` always prevent execution of code in corresponding `setTimeout()`.

Reply:
Ok good to know :)